core/src/main/java/com/datastax/oss/driver/api/core/config/DriverConfigLoader.java

```diff
@@ -200,6 +200,46 @@ static DriverConfigLoader fromUrl(@NonNull URL url) {
         });
   }
 
+  /**
+   * Builds an instance using the driver's default implementation (based on Typesafe config), except
+   * that application-specific options are parsed from the given string.
+   *
+   * <p>The string must be in HOCON format and contain a {@code datastax-java-driver} section.
+   * Options must be separated by line breaks:
+   *
+   * <pre>
+   * DriverConfigLoader.fromString(
+   *         "datastax-java-driver.basic { session-name = my-app\nrequest.timeout = 1 millisecond }")
+   * </pre>
+   *
+   * <p>More precisely, configuration properties are loaded and merged from the following
+   * (first-listed are higher priority):
+   *
+   * <ul>
+   *   <li>the properties in {@code contents}
+   *   <li>{@code reference.conf} (all resources on classpath with this name). In particular, this
+   *       will load the {@code reference.conf} included in the core driver JAR, that defines
+   *       default options for all mandatory options.
+   * </ul>
+   *
+   * Note that this variant does take into account system properties.
```

Comment:
Why?

Reply:
The original use case was tools like Zeppelin that take the configuration from a GUI field, so I thought this would not be applicable.
But you're right, there's no reason to have that arbitrary limitation. I'll add it back.